client/state/data-layer/wpcom/comments/index.js

```diff
@@ -188,6 +191,68 @@ export const announceDeleteFailure = ( action ) => {
 	];
 };
 
+export const emptyComments = ( action ) => ( dispatch ) => {
+	const { siteId, status } = action;
+
+	dispatch(
+		http(
+			{
+				apiVersion: '1',
+				body: {
+					empty_status: status,
+				},
+				method: 'POST',
+				path: `/sites/${ siteId }/comments/delete`,
+			},
+			{
+				...action,
+			}
+		)
+	);
+};
+
+export const handleEmptySuccess = (
+	{ status, siteId, options, refreshCommentListQuery },
+	apiResponse
+) => {
+	const showSuccessNotice = options?.showSuccessNotice;
+
+	return [
+		showSuccessNotice &&
+			successNotice(
+				status === 'spam' ? translate( 'Spam emptied.' ) : translate( 'Trash emptied.' ),
+				{
+					duration: DEFAULT_NOTICE_DURATION,
+					id: 'comment-notice',
+					isPersistent: true,
+				}
+			),
+		!! refreshCommentListQuery && requestCommentsList( refreshCommentListQuery ),
+		{
+			type: COMMENTS_EMPTY_RECEIVE,
+			siteId,
+			status,
+			commentIds: apiResponse?.results?.map( ( x ) => +x ), // convert to number
+		},
+	];
+};
+
+export const announceEmptyFailure = ( action ) => {
+	const { status } = action;
+
+	const error = errorNotice(
+		status === 'spam'
+			? translate( 'Could not empty spam.' )
+			: translate( 'Could not empty trash.' ),
+		{
+			duration: DEFAULT_NOTICE_DURATION,
```

Comment:
Does this also need the same `id` as the success notice?

Perhaps we could reuse the same `noticeOptions` that we're passing to both notices.